Index: orderingservice/fabric/configtx.yaml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/orderingservice/fabric/configtx.yaml b/orderingservice/fabric/configtx.yaml
--- a/orderingservice/fabric/configtx.yaml	(revision ccadcfd08b92fb153d9934c13f9886ce5cd71f9e)
+++ b/orderingservice/fabric/configtx.yaml	(date 1675261864688)
@@ -37,8 +37,9 @@
 
         ID: Org1
 
+        MSPType: idemix
         # MSPDir is the filesystem path which contains the MSP configuration.
-        MSPDir: out/orgs/peerOrganizations/org1.com/msp
+        MSPDir: crypto/peerOrganizations/org1.com
 
         # Policies defines the set of policies at this level of the config tree
         # For organization policies, their canonical path is usually
Index: orderingservice/fabric/enable-idemix
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/orderingservice/fabric/enable-idemix b/orderingservice/fabric/enable-idemix
new file mode 100644
--- /dev/null	(date 1675266789182)
+++ b/orderingservice/fabric/enable-idemix	(date 1675266789182)
@@ -0,0 +1,7 @@
+just init
+idemixgen ca-keygen --output ./crypto/peerOrganizations/org1.com
+idemixgen signerconfig --ca-input ./crypto/peerOrganizations/org1.com --output ./crypto/peerOrganizations/org1.com/users/User1@org1.com --org-unit Org1 --enrollmentId User1 --revocationHandle 1234  --admin
+
+just start
+just listen
+just submit
\ No newline at end of file
Index: utils/connection/security_util.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/utils/connection/security_util.go b/utils/connection/security_util.go
--- a/utils/connection/security_util.go	(revision ccadcfd08b92fb153d9934c13f9886ce5cd71f9e)
+++ b/utils/connection/security_util.go	(date 1675266657096)
@@ -9,13 +9,14 @@
 	"github.com/hyperledger/fabric/msp"
 	mspmgmt "github.com/hyperledger/fabric/msp/mgmt"
 	"github.com/hyperledger/fabric/orderer/common/localconfig"
+	"github.com/spf13/viper"
 	"github.ibm.com/distributed-trust-research/scalable-committer/utils/tls"
 	"google.golang.org/grpc/credentials"
 )
 
 var (
 	projectPath       = os.Getenv("GOPATH") + "/src/github.ibm.com/decentralized-trust-research/scalable-committer/orderingservice/fabric"
-	DefaultCredsPath  = projectPath + "/out/orgs/peerOrganizations/org1.com/peers/peer0.org1.com/"
+	DefaultCredsPath  = projectPath + "/crypto/peerOrganizations/org1.com/users/User1@org1.com/"
 	DefaultRootCAPath = projectPath + "/out/orgs/ordererOrganizations/orderer.org/orderers/raft0.orderer.org/tls/ca.crt"
 	DefaultConfigPath = projectPath
 )
@@ -35,6 +36,7 @@
 
 	os.Setenv("FABRIC_CFG_PATH", configPath)
 	os.Setenv("ORDERER_GENERAL_TLS_ENABLED", "true")
+	viper.Set("peer.localMspType", "idemix") //TODO: Change with env var
 	//os.Setenv("ORDERER_GENERAL_LOCALMSPID", "Org1")
 	//os.Setenv("ORDERER_GENERAL_LOCALMSPDIR", credsPath+"/msp")
 	//os.Setenv("ORDERER_GENERAL_TLS_ROOTCAS", "["+credsPath+"/ca.crt]")
@@ -48,7 +50,7 @@
 		fmt.Println("failed to load config:", err)
 		os.Exit(1)
 	}
-	mspConfig, err := msp.GetLocalMspConfig(credsPath+localMspDir, conf.General.BCCSP, localMspId)
+	mspConfig, err := msp.GetLocalMspConfigWithType(credsPath, conf.General.BCCSP, "Org1", "idemix")
 	if err != nil {
 		fmt.Println("Failed to load MSP config:", err)
 		os.Exit(0)
Index: orderingservice/fabric/testdata/crypto-config.yaml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/orderingservice/fabric/testdata/crypto-config.yaml b/orderingservice/fabric/testdata/crypto-config.yaml
--- a/orderingservice/fabric/testdata/crypto-config.yaml	(revision ccadcfd08b92fb153d9934c13f9886ce5cd71f9e)
+++ b/orderingservice/fabric/testdata/crypto-config.yaml	(date 1675262472583)
@@ -15,15 +15,3 @@
         SANS:
           - 0.0.0.0
           - localhost
-
-PeerOrgs:
-  - Name: Org1
-    Domain: org1.com
-    EnableNodeOUs: true
-    Template:
-      Count: 1
-      SANS:
-        - localhost
-        - 0.0.0.0
-    Users:
-      Count: 1
