diff --git a/orderer/common/msgprocessor/standardchannel.go b/orderer/common/msgprocessor/standardchannel.go
index d98767639..041055806 100644
--- a/orderer/common/msgprocessor/standardchannel.go
+++ b/orderer/common/msgprocessor/standardchannel.go
@@ -11,7 +11,6 @@ import (
 	"github.com/hyperledger/fabric-protos-go/orderer"
 	"github.com/hyperledger/fabric/bccsp"
 	"github.com/hyperledger/fabric/common/channelconfig"
-	"github.com/hyperledger/fabric/common/policies"
 	"github.com/hyperledger/fabric/internal/pkg/identity"
 	"github.com/hyperledger/fabric/protoutil"
 
@@ -67,7 +66,7 @@ func CreateStandardChannelFilters(filterSupport channelconfig.Resources, config
 	rules := []Rule{
 		EmptyRejectRule,
 		NewSizeFilter(filterSupport),
-		NewSigFilter(policies.ChannelWriters, policies.ChannelOrdererWriters, filterSupport),
+		//NewSigFilter(policies.ChannelWriters, policies.ChannelOrdererWriters, filterSupport),
 	}
 
 	if !config.General.Authentication.NoExpirationChecks {
diff --git a/orderer/common/msgprocessor/systemchannel.go b/orderer/common/msgprocessor/systemchannel.go
index fb2f0e873..df77a9b83 100644
--- a/orderer/common/msgprocessor/systemchannel.go
+++ b/orderer/common/msgprocessor/systemchannel.go
@@ -14,7 +14,6 @@ import (
 	"github.com/hyperledger/fabric/bccsp"
 	"github.com/hyperledger/fabric/common/channelconfig"
 	"github.com/hyperledger/fabric/common/configtx"
-	"github.com/hyperledger/fabric/common/policies"
 	"github.com/hyperledger/fabric/internal/pkg/identity"
 	"github.com/hyperledger/fabric/orderer/common/localconfig"
 	"github.com/hyperledger/fabric/protoutil"
@@ -60,7 +59,7 @@ func CreateSystemChannelFilters(
 	rules := []Rule{
 		EmptyRejectRule,
 		NewSizeFilter(ledgerResources),
-		NewSigFilter(policies.ChannelWriters, policies.ChannelOrdererWriters, ledgerResources),
+		//NewSigFilter(policies.ChannelWriters, policies.ChannelOrdererWriters, ledgerResources),
 		NewSystemChannelFilter(ledgerResources, chainCreator, validator),
 	}
 	if !config.General.Authentication.NoExpirationChecks {
