Index: orderer/consensus/smartbft/util.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/orderer/consensus/smartbft/util.go b/orderer/consensus/smartbft/util.go
--- a/orderer/consensus/smartbft/util.go	(revision 695dd57e01c285a43eb297b6744c156d7bb4baea)
+++ b/orderer/consensus/smartbft/util.go	(date 1680764335316)
@@ -238,7 +238,7 @@
 		return config, errors.Wrap(err, "config validation failed")
 	}
 
-	config.RequestMaxBytes = 500 * 1024
+	config.RequestMaxBytes = options.RequestMaxBytes
 	return config, nil
 }
 
